apps/web/src/app/[lang]/_components/__tests__/popular-reviews.test.tsx

```diff
@@ -0,0 +1,117 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { render, screen, within } from '@testing-library/react'
+import { describe, expect, it, vi } from 'vitest'
+import { PopularReviews } from '../popular-reviews'
+
+import * as reviewsService from '@/services/api/reviews'
+import type { FullReview } from '@/services/api/reviews'
+
+vi.mock('@supabase/supabase-js')
+vi.mock('@/context/language', () => {
+  return {
+    useLanguage: () => {
+      return {
+        language: 'en-US',
+        dictionary: {
+          last_week: 'Last week',
+          last_month: 'Last month',
+          all_time: 'All time',
+          popular_reviews: {
+            title: 'Popular reviews',
+            no_reviews_found: "We didn't find any reviews.",
+            explore_popular_movies:
+              'Explore popular movies, and give us your opinion.',
+          },
+        },
+      }
+    },
+  }
+})
+
+describe('PopularReviews component tests', () => {
+  it('should render popular reviews date range options', async () => {
+    vi.spyOn(reviewsService, 'getPopularReviewsService').mockResolvedValue([])
+
+    render(
+      <QueryClientProvider client={new QueryClient()}>
+        <PopularReviews />
+      </QueryClientProvider>,
+    )
+
+    const dateFilterBadges = await screen.findByTestId('date-filter-badges')
+
+    expect(within(dateFilterBadges).getByText(/last week/i)).toBeTruthy()
+    expect(within(dateFilterBadges).getByText(/last month/i)).toBeTruthy()
+    expect(within(dateFilterBadges).getByText(/all time/i)).toBeTruthy()
+  })
+
+  it('should render popular reviews list', async () => {
+    vi.spyOn(reviewsService, 'getPopularReviewsService').mockResolvedValue([
```

Comment:
as I mentioned earlier, not sure how to mock supabase in a better way than this 🫣